cal-itp/benefits
# Refactor: move IdG config to EligibilityVerifier

Closes #2237

## Testing locally

1. Checkout the `main` branch
1. Make sure that your `local_fixtures.json` (or whatever fixture you are using) has a `ClaimsProvider` and an `EligiblityVerifier` that has a foreign key reference to that claims provider.
1. Run `bin/reset_db.sh` (now your local DB mirrors the structure of the `dev` environment)
1. Checkout this PR's branch
1. Run `bin/init.sh` (running this PR's migration on top of existing DB structure)

- [ ] For all `EligiblityVerifier`s that have a `ClaimsProvider`, confirm that the data in `EligiblityVerifier.claims_provider.scope` and `EligiblityVerifier.claims_provider.claim` was migrated to `EligiblityVerifier.claims_scope` and `EligiblityVerifier.claims_claim`, respectively.
- [ ] Confirm that when the code calls an`EligiblityVerifier.claims_scheme` instance, the value of `claims_scheme` equals the value of the associated `ClaimsProvider.scheme` if `claims_scheme` is blank in the database. Otherwise, `claims_scheme` should equal the value set for `EligiblityVerifier` in the database.
- [ ] Flows using scopes/claims from IdG should still work
- [ ] In the admin panel, when not logged in as a super user, check that `Claims scheme` under a selected `Eligibility verifiers` is read-only.

## Notes/Questions

- Is the [setter method](https://github.com/cal-itp/benefits/blob/refactor/move-IdG-config/benefits/core/models.py#L252) for `claims_scheme`
```python
@claims_scheme.setter
def _set_claims_scheme(self, value):
    self._claims_scheme = value
```
actually needed? I included it because I thought that it would be required for setting the value using the admin interface, but I did some testing (with it removed) and it doesn't seem to be needed. We also don't set `claims_scheme` in the code, so maybe I should remove it?
